superset/models/sql_lab.py

```diff
@@ -190,18 +190,17 @@ def columns(self) -> List["TableColumn"]:
             TableColumn,
         )
 
-        columns = []
-        for col in self.extra.get("columns", []):
-            columns.append(
-                TableColumn(
-                    column_name=col["name"],
-                    type=col["type"],
-                    is_dttm=col["is_dttm"],
-                    groupby=True,
-                    filterable=True,
-                )
+        return [
+            TableColumn(
+                column_name=col["name"],
+                database=self.database,
```

Comment:
This is a rewrite of the previous logic using a list comprehension. The only addition is the `database` argument which is required given a SQL Lab `Query` object has no associated `SqlaTable` which is previously were/how the database was obtained.